mods/default/torch.lua

```diff
@@ -1,131 +1,100 @@
-local function on_flood(pos, oldnode, newnode)
-	minetest.add_item(pos, ItemStack("default:torch 1"))
-	-- Play flame-extinguish sound if liquid is not an 'igniter'
-	local nodedef = minetest.registered_items[newnode.name]
-	if not (nodedef and nodedef.groups and
-			nodedef.groups.igniter and nodedef.groups.igniter > 0) then
-		minetest.sound_play(
-			"default_cool_lava",
-			{pos = pos, max_hear_distance = 16, gain = 0.1}
-		)
-	end
-	-- Remove the torch node
-	return false
-end
+function default.register_torch(name, def)
+	local torch = {
+		description = def.description,
+		drawtype = "mesh",
+		tiles = def.tiles,
+		paramtype = "light",
+		paramtype2 = "wallmounted",
+		sunlight_propagates = true,
+		walkable = false,
+		liquids_pointable = false,
+		light_source = def.light_source,
```

Comment:
Consider making `def` a table that overrides these defaults. This way anyone can pass custom definitions, without extending the torch API.